pkg/kubelet/kubelet.go

```diff
@@ -1749,6 +1749,7 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, podStatus *kubecont
 	// Kill pods we can't run.
 	if err := canRunPod(pod); err != nil || pod.DeletionTimestamp != nil {
 		if err := kl.killPod(pod, nil, podStatus); err != nil {
+			err = fmt.Errorf("error killing pod: %v", err)
```

Comment:
1750 declares err, 151 redeclares err, you modify that err, but 1755 returns err from 1750!